src/mono/msbuild/android/build/AndroidApp.targets

```diff
@@ -133,6 +140,12 @@
           Condition="'$(RunAOTCompilation)' == 'true'"
           DependsOnTargets="_AndroidBeforeAotCompileApp">
 
+    <PropertyGroup Condition="'$(_IsLibraryMode)' == 'true'">
+      <_EnableUnmanagedCallersOnlyMethodsExport>true</_EnableUnmanagedCallersOnlyMethodsExport>
+      <_UseAotDataFile>$(UseAotDataFile)<_UseAotDataFile>
```

Comment:
No, I dont have the right logic in autoinit.c to handle `UseAotDataFile == true` right now. Or atleast, it was removed because it wasn't our default use case. I allowed users to specify it if they wanted it and have their own callback logic.

I can add in `mono_install_load_aot_data_hook` thats guarded under an `#if defined(USE_AOT_DATA)` guard. Or should I just remove the flexibility for now?

Reply:
Reintroduced the load aot data hook logic all guarded under a definition that only gets included if there are `.aotdata` files in the compiled assemblies, which only occurs if `UseAotDataFile` is enabled when using the LibraryBuilder.